api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java

```diff
@@ -505,26 +510,39 @@ public List<LeonardoListPersistentDiskResponse> listPersistentDiskByProject(
   public void createApp(CreateAppRequest createAppRequest, DbWorkspace dbWorkspace)
       throws WorkbenchException {
     AppsApi appsApi = appsApiProvider.get();
+    AppType appType = createAppRequest.getAppType();
+
     LeonardoCreateAppRequest leonardoCreateAppRequest = new LeonardoCreateAppRequest();
     Map<String, String> appLabels =
         new ImmutableMap.Builder<String, String>()
-            .put(LeonardoMapper.LEONARDO_LABEL_AOU, "true")
-            .put(LeonardoMapper.LEONARDO_LABEL_CREATED_BY, userProvider.get().getUsername())
-            .put(LeonardoMapper.LEONARDO_LABEL_APP_TYPE, createAppRequest.getAppType().toString())
+            .put(LeonardoLabelHelper.LEONARDO_LABEL_AOU, "true")
+            .put(LeonardoLabelHelper.LEONARDO_LABEL_CREATED_BY, userProvider.get().getUsername())
+            .put(LeonardoLabelHelper.LEONARDO_LABEL_APP_TYPE, appTypeToLabelValue(appType))
             .build();
 
+    LeonardoPersistentDiskRequest diskRequest =
+        leonardoMapper
+            .toLeonardoPersistentDiskRequest(createAppRequest.getPersistentDiskRequest())
+            .labels(
+                upsertLeonardoLabel(
+                    createAppRequest.getPersistentDiskRequest().getLabels(),
+                    LEONARDO_LABEL_APP_TYPE,
+                    appTypeToLabelValue(createAppRequest.getAppType())));
+    // If diskRequest is not set or missing name from request, that means creating new disk.
+    if (diskRequest != null && Strings.isNullOrEmpty(diskRequest.getName())) {
+      diskRequest.setName(userProvider.get().generatePDNameForApp(appType));
+    }
```

Comment:
Product question: Do all apps have PDs?

Are `If diskRequest is not set` and `diskRequest != null` saying different things?

Reply:
If diskRequest is not set is the same as `diskRequest != null`
or missing name from request is `Strings.isNullOrEmpty(diskRequest.getName())`

Reply:
update: Remove diskRequest null check